Coil hackathon branch1

Myoldmopar opened the pull request:
Pull request overview
---------------------
Please change this line to a description of the pull request, with useful supporting information including whether it is a new feature, or fixes a defect, a cross reference to any defects addressed in this PR, the type of changes to be made, and whether diffs are expected/justified based on this change.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

mbadams5 committed:
Update Milo's libraries for dtoa and itoa

Myoldmopar commented:
Thanks for continuing to push on this @rraustad; I look forward to getting back to it shortly! (tomorrow)

rraustad commented:
Comparing to the UnitarySystem_HeatPump example file shows these are very close now. Now on to the other example file.

![newcoilcompare](https://user-images.githubusercontent.com/5778156/36124275-7547bf12-101d-11e8-9782-38e33c779c13.jpg)

Myoldmopar commented:
!!!!!!!

On Mon, Feb 12, 2018 at 16:50 Richard Raustad <notifications@github.com>
wrote:

> Comparing to the UnitarySystem_HeatPump example file shows these are very
> close now. Now on to the other example file.
>
> [image: newcoilcompare]
> <https://user-images.githubusercontent.com/5778156/36124197-2b19b828-101d-11e8-939a-80a8f168ec08.jpg>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/energy-plus/EnergyPlus/pull/26#issuecomment-365090326>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAz3oKbmCHhl3KOTTM5HcHs3EWZgHMEIks5tUMApgaJpZM4R9kab>
> .
>
-- 
Edwin Lee, PhD
Engineer, Commercial Buildings Research Staff
National Renewable Energy Laboratory

rraustad commented:
@Myoldmopar I can't figure out how to get all the speeds to size before this point in code or if it's even possible. The last line below fails on Iter = 2 for the UnitarySystem_VSCoolingCoil_NewDXCoil example file. Around line 2260 of HVACUnitarySystem.

    auto & newCoil = coilCoolingDXs[ UnitarySystem( UnitarySysNum ).CoolingCoilIndex ];
    int const magicNominalModeNum = 0;
    DXCoolCap = newCoil.performance.modes[ magicNominalModeNum ].ratedGrossTotalCap;
    EqSizing.DesCoolingLoad = DXCoolCap;

    for ( Iter = 1; Iter <= UnitarySystem( UnitarySysNum ).NumOfSpeedCooling; ++Iter ) {
        UnitarySystem( UnitarySysNum ).CoolVolumeFlowRate( Iter ) = newCoil.performance.modes[ Iter - 1 ].ratedEvapAirFlowRate;

mjwitte commented:
@rraustad I think the coil performance object has to manage that. It's sizing the nominal now, correct?  When that happens, I think the coil needs to cycle through all of its modes and speeds and set any autosized values right up front so that they are ready to use later.

rraustad commented:
Yes, I agree. I was hoping @Myoldmopar  could add that since we are looking for efficiency here. I would just hack it up. 

The sizeOperatingMode function sizes to the nominal values and then sizeSpeedMode uses those values and fractional values to size each speed.

The mySizeFlag I added wasn't working correctly anyway since it wasn't a vector. These are the types of things I'm not familiar with. I think the calls to the sizing routines need to be moved to a better place, and yes, the entire coil should be sized the first time it's called if the
 !SysSizingCalc flag is true. Once we know the nominal values, all speeds can be sized.

Myoldmopar commented:
I'm focusing on reviewing PRs right now, but I miss working on this - a lot.  I will try to cut some time out today for it, and we can use a little time on the sizing call tomorrow to discuss the state of it and lay out some action items as well.

So for the sizing of each mode/speed, I just need to go through and iterate through all of them, correctly applying the fractional and ratio inputs we laid out?

rraustad commented:
Yes, see sizeSpeedMode, then move that function to the best place and loop through modes and speeds.

rraustad commented:
Here's a rookie question. Why is PsychState a struct instead of a class? Couldn't we just make this a class and part of other classes that use it and just write to it instead of passing it around as an argument and a return?

rraustad commented:
"The only difference between a struct and class in C++ is the default accessibility of member variables and methods. In a struct they are public; in a class they are private."

Myoldmopar commented:
You nailed it. Literally the only difference is that things are by default
public in a struct. So classes that are only easily accessible data members
make nice structs.

rraustad commented:
Great. Change PsycState to a class, inherit it, and stop passing it around?

rraustad commented:
I'm wondering why this isn't faster?

Existing coil model single speed:

    ************* EnergyPlus Completed Successfully-- 0 Warning; 0 Severe Errors; Elapsed Time=00hr 00min 13.66sec

New coil model:

    ************* EnergyPlus Completed Successfully-- 0 Warning; 0 Severe Errors; Elapsed Time=00hr 00min 14.19sec

rraustad commented:
These are release mode annual runs for UnitarySystem_HeatPumpAuto*

Myoldmopar commented:
What's the error file look like for them?  Is the new coil model hitting lots of max iterations or anything?  I wouldn't be surprised if it was since we still have quite a bit of logic to put in there.

rraustad commented:
No max iteration errors and the err file is clean. We don't have any warnings in the new DX coil model yet but since the results match I wouldn't expect coil calculations are slowing things down.

rraustad commented:
Found a difference in pressure used by the new coil model.

    this->inletStateHolder.p = evapInletNode.Press;

While the existing DX cooling coil model uses:

    OutdoorPressure = OutBaroPress;

These values are not the same. Need to track down where node pressure is set and determine method moving forward. Simple method would be to eliminate pressure from PsychState and set as existing coil does.

Also found CBF is different, still need to track down why.

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into CoilSizingFollowup

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into #6485-VRF-power-in-heat-recovery-using-ThermostatOffsetPriority-is-0-when-coils-are-active

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into 6497-Move-setpoint-revision-from-zone-time-step-to-system-time-step

mbadams5 committed:
Begin fixing validation

Myoldmopar committed:
Merge pull request #6514 from NREL/UpdateDataCenter

Fix Followup issues for Enhancements to Data Center Model

Myoldmopar committed:
Merge pull request #6499 from NREL/6497-Move-setpoint-revision-from-zone-time-step-to-system-time-step

Move setpoint calculation from zone time step to system time step

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into CoilSizingFollowup

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into #6485-VRF-power-in-heat-recovery-using-ThermostatOffsetPriority-is-0-when-coils-are-active

Myoldmopar committed:
Merge pull request #6500 from NREL/CoilSizingFollowup

Change remaining Coil:Heating:Gas to Fuel plus cleanups on new coil sizing reports

Myoldmopar committed:
Merge pull request #6498 from NREL/#6485-VRF-power-in-heat-recovery-using-ThermostatOffsetPriority-is-0-when-coils-are-active

VRF power in heat recovery using thermostat offset priority is 0 when coils are active

lgu1234 committed:
Improve setpoint calculation

Myoldmopar committed:
Protect the extra load calculation to only const-speed-branch-pump loopsides

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into BranchPumpsAgain

Myoldmopar committed:
Merge pull request #6523 from NREL/BranchPumpsAgain

Continuation of Constant Speed Branch Pump Fixes

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into 6464-fuel-cell-solver

Myoldmopar committed:
Fix license text

Myoldmopar committed:
Prep version numbers, dates, and readme for release

Myoldmopar committed:
Merge pull request #6542 from NREL/6464-fuel-cell-solver

6464 Fix Fuel Cell Solver Bounds

mbadams5 committed:
Working validation and refactored getObjectItem

lgu1234 committed:
Address comments

lgu1234 committed:
Merge branch 'develop' into Improve-setpoint-calculation

Myoldmopar committed:
Merge pull request #6541 from NREL/Improve-setpoint-calculation

Improve setpoint calculation

Myoldmopar committed:
Fix up file problems identified by @mjwitte

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into PrepFor89

mjwitte committed:
CBVAV fix 6544 and unit test

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into CBVAVFix6544

mjwitte committed:
InitZoneAirSetPoints protect against zone with no surfaces

mbadams5 committed:
Fix failing unit and integration tests

mbadams5 committed:
Allow cbor/msgpack hidden input again

mbadams5 committed:
Update json library

mbadams5 committed:
Merge branch 'develop' into json_followup

mbadams5 committed:
Clean up code

mbadams5 committed:
Merge branch 'develop' into json_followup

# Conflicts:
#	tst/EnergyPlus/unit/InputProcessor.unit.cc

mbadams5 committed:
Fix merge conflict

# Conflicts:
#	tst/EnergyPlus/unit/InputProcessor.unit.cc

mjwitte committed:
CBVAV fix build warnings

mbadams5 committed:
Fix build error and a couple err file diffs

mjwitte committed:
CBVAV add clear_state

Myoldmopar committed:
Merge pull request #6543 from NREL/PrepFor89

EnergyPlus 8.9.0 Release Preparation

Myoldmopar committed:
Merge pull request #6547 from NREL/CBVAVFix6544

Fix CBVAV terminal unit error introduced with multi-HVAC controls

mbadams5 committed:
Address build warning and review comment

mbadams5 committed:
Merge branch 'develop' into json_followup

Myoldmopar committed:
Add version mismatch warning if input errors exist

mbadams5 committed:
Fix linux build error and parametric processor input error

Myoldmopar committed:
Merge branch 'json_followup' of github.com:NREL/EnergyPlus into json_followup

Myoldmopar committed:
Fix piping system unit test duplicate name issue

Myoldmopar committed:
Temporarily make the duplicate name into warnings

mbadams5 committed:
Fix failing unit test

mbadams5 committed:
Merge branch 'json_followup' of https://github.com/NREL/EnergyPlus into json_followup

mbadams5 committed:
Fix final failing unit test

Myoldmopar committed:
Fix bad choice fields in performance test files

Myoldmopar committed:
Merge pull request #6548 from NREL/json_followup

JSON followup fixes

Myoldmopar committed:
Duplicate names are errors, skip error handling until json validation

mjwitte committed:
DataSet unit tests - fix refr cases punt refr compressors

mjwitte committed:
Input validation - add value string to error msg

mjwitte committed:
Input validation - fix unit tests to match revised error msg

Myoldmopar committed:
Fix up duplicate chiller object names

jasondegraw committed:
Correct program install placement, CMake message

jasondegraw committed:
Merge branch 'PrepFor89Again' of https://github.com/NREL/EnergyPlus into PrepFor89Again

mjwitte committed:
Windows ep-launch save epJSONout

mjwitte committed:
Merge branch 'PrepFor89Again' of https://github.com/NREL/EnergyPlus into PrepFor89Again

Myoldmopar committed:
Fix precip duplicate name test failure

Myoldmopar committed:
Merge branch 'PrepFor89Again' of github.com:NREL/EnergyPlus into PrepFor89Again

mjwitte committed:
Input validation - fix more unit tests

mjwitte committed:
Merge branch 'PrepFor89Again' of https://github.com/NREL/EnergyPlus into PrepFor89Again

Myoldmopar committed:
Fixup readme for release

Myoldmopar committed:
Merge branch 'PrepFor89Again' of github.com:NREL/EnergyPlus into PrepFor89Again

Myoldmopar committed:
Fix line endings in unit test error string comparison

mjwitte committed:
Input validation - fix unit test line endings

mbadams5 committed:
Address bad encoding causing missing table entries

mbadams5 committed:
Package epJSON schema

Myoldmopar committed:
Merge pull request #6564 from NREL/PrepFor89Again

Final approach on 8.9.0

lefticus committed:
Place formatting files specifically in src dir

lefticus committed:
Add additional information on formatting

lefticus committed:
Tweak .clang-format to minimize changes, update readme

energyplus-style-bot committed:
Merge remote-tracking branch 'origin/Update_Style_Guides' into develop

energyplus-style-bot committed:
Apply formatting with clang-format

energyplus-style-bot committed:
Fix header includes found by reordering

lefticus committed:
Add verify_file_formatting file

energyplus-style-bot committed:
Apply formatting to tst folder

energyplus-style-bot committed:
Specify BinPackArguments: false

This should make some reformatting and transition easier for those who
have their IDEs configured to use the .clang-format file

Myoldmopar committed:
Fix up the output variable parser with regex, comment changes in two source files

Myoldmopar committed:
Add op schemes to plant

mbadams5 committed:
Update re2 to 2018-03-01

mbadams5 committed:
Update SQLite to 3.22.0

mbadams5 committed:
Update GTest to 1.8.0

mbadams5 committed:
Disable install output for new gtest and re2

mbadams5 committed:
Update eigen library to 3.3.4

mbadams5 committed:
Fix linux build error

mbadams5 committed:
Fixed build errors

mbadams5 committed:
Fix linux build issue

mjwitte committed:
New DX cooling coil protect against div by zero

mjwitte committed:
New DX cooling coil transition SingleSpeed

mjwitte committed:
New DX cooling coil fix array bounds and div by zero

mjwitte committed:
New DX cooling coil transition VariableSpeed

mjwitte committed:
New DX cooling coil transition fix and new singlespeed test file

mjwitte commented on src/EnergyPlus/Coils/CoilCoolingDXCurveFitSpeed.cc:
```diff
@@ -0,0 +1,378 @@
+
+void CoilCoolingDXCurveFitSpeed::instantiateFromInputSpec( CoilCoolingDXCurveFitSpeedInputSpecification input_data, CoilCoolingDXCurveFitOperatingMode * _parentMode ) {
+    this->original_input_specs = input_data;
+    this->parentMode = _parentMode;
```
@Myoldmopar Running with new test file HVACTemplate-5ZoneUnitarySystem-SingleSpeed-exp.idf which has 5 unitary systems and 5 dx coils. At this point, coil1 speed 1 "Sys 1 Furnace DX Cool Cooling Coil Speed 1 Performance" thinks its parent is the correct one "Sys 1 Furnace DX Cool Cooling Coil Operating Mode".

mjwitte commented on src/EnergyPlus/Coils/CoilCoolingDXCurveFitSpeed.cc:
```diff
@@ -0,0 +1,378 @@
+
+    std::string RoutineName = "sizeSpeedMode";
+
+    this->rated_total_capacity = this->original_input_specs.gross_rated_total_cooling_capacity_ratio_to_nominal * parentMode->ratedGrossTotalCap;
```
But, by the time it gets here, parentMode is now pointing to "Sys 5 Furnace DX Cool Cooling Coil Operating Mode" (should be Sys 1 . . . ) and the sizing fails further down, because coil Sys 5 operating mode hasn't been sized yet.

mjwitte committed:
New DX cooling coil transition TwoSpeed

mjwitte committed:
New DX cooling coil add Fuel Type field to idd

mjwitte committed:
New DX cooling coil transition MultiSpeed and add Fuel Type field

mjwitte commented on idd/Energy+.idd.in:
```diff
@@ -43676,6 +43676,424 @@ Coil:Cooling:Water:DetailedGeometry,
+  A10, \field Evaporative Condenser Supply Water Storage Tank Name
+       \type object-list
+       \object-list WaterStorageTankNames
+  A11; \field Fuel Type
```
@rraustad @Myoldmopar @nealkruis While writing transition code for the multispeed coil, I realized we didn't have a field for Fuel Type in the new coil. I've added it to the top level Coil:Cooling:DX but wondering if it should go in Coil:Cooling:DX:CurveFit:Performance instead.

rraustad replied:
Wherever the energy use is calculated. I can't recall right now but I'm fairly sure it's higher than the speed level.

mjwitte replied:
Currently, energy use is set [here](https://github.com/energy-plus/EnergyPlus/pull/26/files#diff-a8f6d2c681d1b4c0251e967b1ef4de1fR134) in `CoilCoolingDX::simulate` using power that was computed further down. However, there really needs to be two separate power variables tracked, one for the compressor power (which could be any fuel type) and one for electrical parasitics including controls, basin heater, crankcase heater and condenser fan. We may need more inputs to fully support this. And the current code isn't calculating crankcase heater and basin heater power use - but the inputs for those are in Coil:Cooling:DX:CurveFit:Performance presently, so that's probably where Fuel Type should go.

mjwitte committed:
New DX cooling coil move fuel type to Performance object

jasondegraw committed:
Change error call to use the higher level routine

mjwitte committed:
New DX cooling coil fix speed object parent pointer problem

mjwitte committed:
New DX cooling coil fix condenser node connection problem

mjwitte committed:
New DX cooling coil revert main.cc debug

mjwitte commented on src/EnergyPlus/Coils/CoilCoolingDXCurveFitOperatingMode.cc:
```diff
@@ -0,0 +1,170 @@
+    // Set speed objects parent pointer (this is kind of a test - probably needs to be a separate init function)
+    for (auto & curSpeed : this->speeds) {
+        curSpeed.parentMode = this;
+    }
```
@Myoldmopar Moved assignment of parentMode to here so that it's done after the operating mode object has been fully built and emplaced. Seems to solve the moving pointer problem and gets past the sizing error. Not sure if this is the best spot to put this.

mjwitte committed:
New DX cooling coil transition MultiStageWithHumidityControlMode

mjwitte committed:
New DX cooling coil transition move PreProccess msgs to warnings

Myoldmopar committed:
Update README.md

Myoldmopar committed:
Merge pull request #6740 from NREL/InvestigateCI

Dummy PR to get develop moving again

lgu1234 committed:
Upload a fix

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into FixSetupOutputVariableScript

lgu1234 committed:
Upload a fix

Myoldmopar committed:
Merge pull request #6608 from NREL/FixSetupOutputVariableScript

Output Variable Parser Fix

lgu1234 committed:
Address overshooting using correct zone air temperature at previous time step

lgu1234 committed:
Merge branch 'develop' into 157643005-ddress-limitations-and-overshooting-problem-using-ZTM1

Myoldmopar committed:
Update readme organization and badges

Myoldmopar committed:
Fix color

Myoldmopar committed:
Reduce enormous line lengths on readme

Myoldmopar committed:
Tidied up dashboard discussion

Myoldmopar committed:
Merge pull request #6755 from NREL/ReadmeUpdates

Update readme organization and badges

Myoldmopar committed:
Merge pull request #6721 from NREL/change-error-call

Change error call to use the higher level routine

Nigusse committed:
corrected moisture sorption curve slope

Nigusse committed:
EMPD Model EngRef Equation Fix

Nigusse committed:
Added Unit Test for EMPD model moisture sorption Equation slope Fix

RKStrand committed:
Correction of bug

The ZoneSupplyAirTemp was being used as a reference temperature when
the coefficients in the calculation procedures were using the
ZoneMeanAirTemp.  This was fixed for all of the Fisher-Pedersen models
as requested by the GitHub issue.

lgu1234 committed:
Upload two revised example files

RKStrand committed:
Addition of unit tests

Addition of unit tests to test corrected code.

rraustad committed:
Retail -999 and -99999 in Coil Sizing Summary report

rraustad committed:
Add Zone Name to Coil Sizing Details table

rraustad committed:
Add plant initialization for plant sizing.

rraustad committed:
Correct zone coil inlet conditions for single duct TUs and try new zone name method.

rraustad committed:
Add unit test, other cleanup.

rraustad committed:
Include fan coil inlet conditions

lgu1234 committed:
Add re-initialization statements in the clear-state function

lgu1234 committed:
Merge branch 'develop' into 157519812-Multiple-CO2-controllers

lgu1234 committed:
Add re-initialization statements in the clear-state function

lgu1234 committed:
Merge branch 'develop' into 157599603-Multiple-GenericContam-controllers

lgu1234 committed:
Assign a member variable value from inputs

Myoldmopar committed:
Merge pull request #6745 from NREL/157519812-Multiple-CO2-controllers

Set multiple CO2 controllers correctly

Myoldmopar committed:
Merge develop, try to resolve conflict

Myoldmopar committed:
Merge pull request #6751 from NREL/157599603-Multiple-GenericContam-controllers

Set multiple GenericContam controllers correctly

lgu1234 committed:
Address comments

lgu1234 committed:
Merge branch 'develop' into 157643005-ddress-limitations-and-overshooting-problem-using-ZTM1

Commit:
Remove extra space at end of readvars header

mbadams5 committed:
Merge branch 'develop' into update-libraries

mbadams5 committed:
Update SQLite to 3.24.0

lgu1234 committed:
Upload a fix

lgu1234 committed:
Remove a warning and cleanup

rraustad committed:
Rookie try at a fix

rraustad committed:
Revert previous change and correct int return for name not found

rraustad committed:
Add unit tests

lgu1234 committed:
Upload a fix

lgu1234 committed:
Issue a fatal error with two or more fans in an AirLoop using AirflowNetwork model (#6758)

* Upload a fix

* Address comments

RKStrand committed:
Merge branch 'develop' into 157485520-FisherPedersenConvectEqRefTempBug

RKStrand committed:
Correct formatting warnings

Tabs were being used to indent lines.  This has now changed to spaces
as per the formatting requirements.  Should eliminate the warning
messages.

lgu1234 committed:
Upload a fix

lgu1234 committed:
Merge branch 'develop' into 158218302-SolveRoot-RegulaFalsi-fails-with-an-underflow-in-VRF-model-#6515

RKStrand committed:
Merge branch 'develop' into #6763-Swimming-pool-flow-not-added-to-plant-loop-sizing

lgu1234 committed:
Remove a warning

JasonGlazer committed:
Merge pull request #6773 from NREL/158148155-Extra-fields-in-ConstructionComplexFenestrationState-causes-crash-#6744

Issue severe error for vertex size mismatch between base and boundary surfaces

Myoldmopar committed:
Apply clang format to all E+ source files

mjwitte committed:
Merge pull request #6783 from NREL/ApplyClangFormat

Apply clang format to all E+ source files

lgu1234 committed:
Initial fix

lgu1234 committed:
Merge branch 'develop' into 158368465-Improve-error-#6375

RKStrand committed:
Merge branch 'develop' into #6763-Swimming-pool-flow-not-added-to-plant-loop-sizing

# Conflicts:
#	src/EnergyPlus/SwimmingPool.cc

RKStrand committed:
Merge branch 'develop' into 157485520-FisherPedersenConvectEqRefTempBug

Myoldmopar committed:
Merge branch 'develop' into #6763-Swimming-pool-flow-not-added-to-plant-loop-sizing

Myoldmopar committed:
Merge pull request #6768 from NREL/157889248-Issue6507

Minimum OA dry bulb is not properly processed

Myoldmopar committed:
Adding in command line argument flag for fixing this header

Myoldmopar committed:
Add command line argument to IO ref ReadVars documentation

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 157643005-ddress-limitations-and-overshooting-problem-using-ZTM1

lgu1234 committed:
Validate zone inlet nodes and provide error messages if not found

lgu1234 committed:
Merge branch 'develop' into 158368465-Improve-error-#6375

lgu1234 committed:
Remove unused variables and cleanup

rraustad committed:
Correct crash and create unit test

rraustad committed:
Add better comments in unit test

lgu1234 committed:
Upload a fix

RKStrand committed:
Merge branch 'develop' into #6763-Swimming-pool-flow-not-added-to-plant-loop-sizing

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into 179D_Reporting-v8.8.0

Myoldmopar committed:
Merge pull request #6760 from NREL/157741082-Remove-unnecessary-warnings-6710

Remove unnecessary warnings in two example files

Myoldmopar committed:
Merge pull request #6633 from NREL/v8.9PlusIDDFixes

Fix Tiny IDD Issues

Myoldmopar committed:
Merge pull request #6759 from NREL/157817954_Issue#6634

EMPD model moisture sorption curve slope equation is incorrrect

Myoldmopar committed:
Merge pull request #6769 from NREL/ReadVarsHeader

Remove extra space at end of readvars header

Myoldmopar committed:
Merge pull request #6776 from NREL/#158218232-Underflow-in-CalcFluidHeatExchanger-#6516

Setup a correct value for an exp function with underflow value

Myoldmopar committed:
Merge pull request #6786 from NREL/158368465-Improve-error-#6375

Improve error message when SetpointManager:SingleZone:Reheat does not find matching zone inlet node

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into 179D_Reporting-v8.8.0

Myoldmopar committed:
Merge pull request #6779 from NREL/158218302-SolveRoot-RegulaFalsi-fails-with-an-underflow-in-VRF-model-#6515

Avoid underflow in SolveRoot

RKStrand committed:
Merge branch 'develop' into #6763-Swimming-pool-flow-not-added-to-plant-loop-sizing

RKStrand committed:
Merge remote-tracking branch 'origin/#6763-Swimming-pool-flow-not-added-to-plant-loop-sizing' into #6763-Swimming-pool-flow-not-added-to-plant-loop-sizing

# Conflicts:
#	src/EnergyPlus/SwimmingPool.cc

RKStrand committed:
Committing again

conflicts keep showing up when none should exist

RKStrand committed:
Merge branch 'develop' into #6763-Swimming-pool-flow-not-added-to-plant-loop-sizing

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into 179D_Reporting-v8.8.0

Myoldmopar committed:
Comment out broken unit test temporarily

rraustad committed:
Merge branch 'develop' of https://github.com/NREL/EnergyPlus into #6598-FanSystemModel-crashes-using-Continuous-speed-control-method

Nigusse committed:
Hot Water Initialization Fix First Round (Issue #5821)

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 157643005-ddress-limitations-and-overshooting-problem-using-ZTM1

Myoldmopar committed:
Disable OA unit test file for now, just to see if I can get the builds clean again

Myoldmopar committed:
Merge pull request #6811 from NREL/HotFix-FailingHXUnitTest

Disable broken unit test

Myoldmopar committed:
Get the core stuff merged in, dont run the test yet

Nigusse committed:
Merge branch 'develop' of https://github.com/NREL/EnergyPlus into 158868652_Issue#5821

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into 179D_Reporting-v8.8.0

Nigusse committed:
Replaced 5.0C with CWInitConvTemp (Issue #5821)

RKStrand committed:
Merge branch 'develop' into #6763-Swimming-pool-flow-not-added-to-plant-loop-sizing

Nigusse committed:
Added Unit Tests (Issue #5821)

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 157485520-FisherPedersenConvectEqRefTempBug

lgu1234 committed:
Address comments

lgu1234 committed:
Merge branch '157643005-ddress-limitations-and-overshooting-problem-using-ZTM1' of https://github.com/NREL/EnergyPlus into 157643005-ddress-limitations-and-overshooting-problem-using-ZTM1

lgu1234 committed:
Merge branch 'develop' into 157643005-ddress-limitations-and-overshooting-problem-using-ZTM1

mbadams5 committed:
Change Schedule:Year to extensible, but limit number of items

Fixes #6729

mbadams5 committed:
Merge pull request #6775 from NREL/#6774-getObjectItemNum-does-not-differentiate-between-object-type-and-name

Misspelled object names result in warning that says object type is wrong

mbadams5 committed:
Merge remote-tracking branch 'origin/develop' into update-libraries

RKStrand committed:
Unit test

Addition of unit test and formatting changes (deleted many old
unnecessary comment lines)

RKStrand committed:
Merge branch 'develop' into #6763-Swimming-pool-flow-not-added-to-plant-loop-sizing

RKStrand committed:
Merge branch 'develop' into 157485520-FisherPedersenConvectEqRefTempBug

mjwitte committed:
Merge pull request #6754 from NREL/157643005-ddress-limitations-and-overshooting-problem-using-ZTM1

Address overshooting using correct zone air temperature at previous timestep

mbadams5 committed:
Update re2 library to slightly older version for gcc 4.8.4

mbadams5 committed:
Fix issue when Output:Variable key uses regex symbols

This happens when using a regex symbol but wanting to perform a full key comparison and not actually use regex.

mbadams5 committed:
Merge branch 'develop' into issue_6729

mbadams5 committed:
Merge branch 'develop' into update-libraries

Myoldmopar committed:
Disable platform for now, get the code in, and turn it back on next week

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into 179D_Reporting-v8.8.0

mbadams5 committed:
Address overwritten objects with blank Names

This issue was partially address before 8.9 release, however, not all
cases were covered. This should fully address the issue now.

mbadams5 committed:
Fix issue causing diffs and integration failure

Some objects REQUIRE the name to be blank if using duplicates. Only special case RunPeriod for now.

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 157485520-FisherPedersenConvectEqRefTempBug

Myoldmopar committed:
Fix allocation issue for metered output variables

Myoldmopar committed:
Add unit test to cover case when the metered output var is the longest EMS object

rraustad committed:
Merge pull request #6765 from NREL/#6763-Swimming-pool-flow-not-added-to-plant-loop-sizing

Add plant initialization for SwimmingPool:Indoor.

mjwitte committed:
New DX cooling coil fixes for multimode and hx assisted

mjwitte committed:
New DX cooling coil transition fixes for 2-speed DX

mjwitte committed:
New DX cooling coil transition test files

mjwitte commented:
@Myoldmopar et al I've added some folders in [testfiles/NewDXCoilTransitionTests](https://github.com/energy-plus/EnergyPlus/tree/CoilHackathonBranch1/testfiles/NewDXCoilTransitionTests) to test this with unitary system and a variety of DX coil types. I may have missed some. All of these run except HVACTemplate-5ZoneUnitarySystem-HXAssistedDX-exp because coilsystem doesn't recognize the new coil yet. And the transition code is still in v8.8tov8.9.

nealkruis commented on src/EnergyPlus/Coils/CoilCoolingDXCurveFitSpeed.cc:
```diff
@@ -261,8 +259,41 @@ Psychrometrics::PsychState CoilCoolingDXCurveFitSpeed::CalcSpeedOutput( Psychrom
+  const std::size_t maxIter(50);
+
+  while( (iter <= maxIter) ) {
+
```
@Myoldmopar This is where I was when I was re-writing the bypass factor calculation. You can reference the CalcBypassFactor method in either DXCoils or VariableSpeedCoils (though they are slightly different so you'll want to take the best from both).

mjwitte committed:
Merge pull request #6764 from NREL/157485520-FisherPedersenConvectEqRefTempBug

Correction of Reference Temperature for Fisher-Pedersen Convection Models when Specified for Surfaces

Myoldmopar committed:
Merge pull request #6461 from NREL/179D_Reporting-v8.8.0

179d reporting based on v8.8.0

Myoldmopar committed:
Bulk changes for version increment, still need to update file version ids

Myoldmopar committed:
Increment version ids for test files, datasets, and unit tests

Myoldmopar committed:
Add input/output rules files along with updated install cmake stuff

Myoldmopar committed:
Fix last two test files that showed version errors

Myoldmopar committed:
Merge pull request #6836 from NREL/PrepFor90

Prep for 9.0

Myoldmopar committed:
Merge pull request #6671 from NREL/update-libraries

Update libraries

Myoldmopar committed:
Merge pull request #6828 from NREL/BugFix-EMSAlphaAllocation

Fix allocation issue for metered output variables, Fixes #6827

Myoldmopar committed:
Merge pull request #6824 from NREL/issue_6802

Fix issue when Output:Variable key uses regex symbols for full key comparison

Myoldmopar committed:
Merge pull request #6823 from NREL/issue_6729

Address epJSON conversion issue with Schedule:Year

Myoldmopar committed:
Merge pull request #6822 from NREL/issue_6736

Address overwritten RunPeriod objects

Myoldmopar committed:
Merge pull request #6813 from NREL/158868652_Issue#5821

Heating water coil sizing mixing up hot and cold initialization temperatures

Myoldmopar committed:
Merge pull request #6805 from NREL/#6598-FanSystemModel-crashes-using-Continuous-speed-control-method

Correct Fan:SystemModel crash when Continuous mode is used without power as a function of flow fraction curve

Myoldmopar committed:
Merge pull request #6807 from NREL/158538153-Maximum-Curve-Output-is-always-ignored-#6365

Assign correct input values for Curve:ExponentialSkewNormal

Myoldmopar committed:
Merge develop, resolve conflict with CW->HW init temp

Myoldmopar committed:
Merge pull request #6762 from NREL/#6761-Coil-Sizing-Summary-shows-incorrect-VAV-box-coil-inlet-air-conditions

Correct terminal unit heating coil inlet air conditions used for sizing and/or reporting.

Myoldmopar committed:
Merged develop, all conflicts resolved except one

Myoldmopar committed:
Fixed input processor changes

Myoldmopar committed:
Fix some conflict markers

Myoldmopar committed:
Applied style again
